### PR TITLE
Add `hub.customScopes` directly mapping to `c.JupyterHub.custom_scopes`

### LINKED ISSUE #3265: Add support for `custom_scopes`
### Proposed change

Add support for `custom_scopes` as described here:
- https://jupyterhub.readthedocs.io/en/stable/rbac/scopes.html#custom-scopes

Allowing to configure those directly in the chart instead of using `extraConfig` would be less error prone. 

### Who would use this feature?

Users using `custom_scopes`

### PR DESCRIPTION
**Pull request by Ph0tonic**
This is a first draft to support `custom_scopes`, I am not sure how to test it. Any feedback and help is welcome. 

Fixes #3265

\cc @consideRatio and @minrk

### CODE REVIEW

File: jupyterhub/values.schema.yaml [hunk 1]
```diff
@@ -1410,6 +1410,31 @@ properties:
...
+              "custom:myservice:read":
+                description: read-only access to myservice
+
+              "custom:myservice:write":
+                description: write access to myservice 
+                subscopes: ["custom:myservice:read"]
+          ```
+
+          When configuring JupyterHub custom scopes via this Helm chart, the `name`
+          field can be omitted as it can be implied by the dictionary key.
```
Comment (consideRatio) [inline jupyterhub/values.schema.yaml:1437]: There doesn't seem to be a `name` field for custom_scopes, instead its the dictionary key. So while `JupyterHub.load_roles` is a list and `hub.loadRoles` is a dict translated to a list, the `JupyterHub.custom_scopes` is a dict and doesn't need a translation.

This also means that we can without issues configure this directly via `hub.config.JupyterHub.custom_scopes` as well without providing a dedicated option for this I think.

@Ph0tonic could you update this description to not mention the `name` field?

**Comment by consideRatio** [comment]
@minrk do you think z2jh should duplicate this, or ask that users rely on `hub.config.JupyterHub.custom_scopes` ? I think this makes it more findable, but a bit less sustainable.

**Comment by manics** [comment]
I think we should encourage use of passthrough as it reduces maintenance. We should update the docs though, there's still a warning `Currently intended only for auth config`:
https://z2jh.jupyter.org/en/stable/resources/reference.html#hub-config

**Comment by Ph0tonic** [comment]
Hello,
Thanks for your informations.
Not sure however to have understood everything. So the idea is to directly use :
```yaml
hub:
  config:
    JupyterHub:
      custom_scopes:
        ...
```

So shall I close this issue ? And update the documentation ?

**Comment by consideRatio** [comment]
> So the idea is to directly use :
> 
> ```yaml
> hub:
>   config:
>     JupyterHub:
>       custom_scopes:
>         ...
> ```

Yes, but I'm not sure where to document that use of `hub.config` passthrough should be used for `JupyterHub.custom_scopes` but not for `hub.loadRoles` for example. Hmmm...

**Comment by manics** [comment]
I think we can either say
- use `hub.config` where there is no explicit chart config (easiest for us)
- or list the properties than can be set with `hub.config` (more work as we need to keep it up to date)

**Comment by consideRatio** [comment]
I opened https://github.com/jupyterhub/zero-to-jupyterhub-k8s/issues/3271, I think the path forward is to settle for updating docs about `hub.config`, and that there is no remaining action point here.

**Pull request closed without merging**